src/hb-buffer-serialize.cc

```diff
@@ -379,6 +449,117 @@ hb_buffer_serialize_glyphs (hb_buffer_t *buffer,
   }
 }
 
+unsigned int
+hb_buffer_serialize_unicode (hb_buffer_t *buffer,
+					unsigned int start,
+					unsigned int end,
+					char *buf,
+					unsigned int buf_size,
+					unsigned int *buf_consumed,
+					hb_buffer_serialize_format_t format)
+{
+	assert (start <= end && end <= buffer->len);
+
+	unsigned int sconsumed;
+	if (!buf_consumed)
+		buf_consumed = &sconsumed;
+	*buf_consumed = 0;
+	if (buf_size)
+		*buf = '\0';
+
+	assert ((!buffer->len && (buffer->content_type == HB_BUFFER_CONTENT_TYPE_INVALID)) ||
+		(buffer->content_type == HB_BUFFER_CONTENT_TYPE_UNICODE));
+
+	if (unlikely (start == end))
+		return 0;
+
+	switch (format)
+	{
+		case HB_BUFFER_SERIALIZE_FORMAT_TEXT:
+			return _hb_buffer_serialize_unicode_text (buffer, start, end,
+								 buf, buf_size, buf_consumed);
+
+		case HB_BUFFER_SERIALIZE_FORMAT_JSON:
+			return _hb_buffer_serialize_unicode_json (buffer, start, end,
+								 buf, buf_size, buf_consumed);
+
+		default:
+		case HB_BUFFER_SERIALIZE_FORMAT_INVALID:
+			return 0;
+
+	}
+}
+/**
+ * hb_buffer_serialize:
+ * @buffer: an #hb_buffer_t buffer.
+ * @start: the first item in @buffer to serialize.
+ * @end: the last item in @buffer to serialize.
+ * @buf: (out) (array length=buf_size) (element-type uint8_t): output string to
+ *       write serialized buffer into.
+ * @buf_size: the size of @buf.
+ * @buf_consumed: (out) (allow-none): if not %NULL, will be set to the number of byes written into @buf.
+ * @font: (allow-none): the #hb_font_t used to shape this buffer, needed to
+ *        read glyph names and extents. If %NULL, and empty font will be used.
+ * @format: the #hb_buffer_serialize_format_t to use for formatting the output.
+ * @flags: the #hb_buffer_serialize_flags_t that control what glyph properties
+ *         to serialize.
+ *
+ * Serializes @buffer into a textual representation of its content, whether
+ * Unicode codepoints or glyph identifiers and positioning information. This is
+ * useful for showing the contents of the buffer, for example during debugging.
+ * There are currently two supported serialization formats:
+ *
+ * ## text
```

Comment:
The description if the format should not be repeated here, it should instead refer to the documentation of the other two functions.